how-to/authentication/placeos-api-auth.md

```diff
@@ -0,0 +1,133 @@
+---
+title: PlaceOS API Authentication
+description: Steps required for authenticating against the PlaceOS API
+sidebar_position: 6
+---
+
+Access to PlaceOS is secured via [OAuth2](https://auth0.com/docs/authorization/which-oauth-2-0-flow-should-i-use#is-the-client-a-native-mobile-app-).
+
+Before interacting, your app or integration will need to authenticate and obtain a valid access token.   
+Once authenticated, this token must accompany all requests.
+
+This can either be included as an `Authorization` header (recommended):
+
+```
+http example.com/api/control/systems 'Authorization:bearer <access token>'
+```
+
+Or, as a query parameter:
+
+```
+http example.com/api/control/systems bearer_token==<access token>
+```
+
+## Registering Your Application
+
+All applications using the PlaceOS API need to be registered in Backoffice. 
+Details on this can be found in the [Add an Application](../backoffice/add-application.md) guide.
+
+Once registered, take note of the: 
+    * `client_id` 
+    * `client_secret` 
+    * `redirect_uri` (optional based on the auth flow your application will use)
+
+## Obtaining An Access Token
+
+Using OAuth2, a few approaches are available to obtain an access token. 
+
+Here's some recommendations based on common API uses:
+
+|Auth Flow |Recommended Use Case |
+|---|---|
+Implicit|Web, mobile and desktop apps communicating directly with the PlaceOS API.|
```

Comment:
The [OAuth 2.0 Security Best Current Practice document recommends against using the Implicit flow entirely](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-2.1.2), and OAuth 2.0 for Browser-Based Apps describes the technique of using the authorization code flow with PKCE instead.

That said, it's only an issue with leakage of the access token (and the inability to detect replay attacks with that token) and probably not something that's always going to affect clients, especially in more closed off environments, just thought I should mention it.